### PR TITLE
CallSupport: allow calls on derived classes

### PR DESCRIPTION
**Pull request by merlinND**
The DrJit registry remains untouched, we simply allow instantiating a `CallSupport<Child>` which exposes the interface of  both `Child` and its parent class.

### CODE REVIEW

File: docs/cpp.rst [hunk 1]
```diff
@@ -322,6 +322,20 @@ on a Dr.Jit instance arrays. Below is an overview of the available macros:
...
    Demarcates the end of an interface block.
 
+.. c:macro:: DRJIT_CALL_TEMPLATE_INHERITED_BEGIN(Name, Parent)
+
+   A variant of the above macro that should be used when ``Name`` refers to a
+   template class that inherits from another template class ``Parent``.
+   The class ``Name`` should only be registered under the registry domain of ``Parent``.
+   The interface of ``Parent`` must have already been exposed with a call to
+   ``DRJIT_CALL_TEMPLATE_BEGIN(Parent) ... DRJIT_CALL_END()``.
+   The interface will then be extended with any methods listed in this block.
```
Comment (wjakob) [inline docs/cpp.rst:332]: Could you add that it's only safe to use this feature in polymorphic types, where C++ can use RTTI to determine if a particular type is a subclass of `Parent` or `Name`?

**Review by wjakob — COMMENTED** [review]
Could I ask you add tests in the Dr.Jit repository? There is a file `tests/call_ext.cpp` and associated Python script that tests the functionality in a self-contained way here without depending on Mitsuba.

**Comment by merlinND** [comment]
Added this feature to the test bindings. I'll squash the commit history before merging once CI is green, and let me know if you'd like any other change!

**Comment by wjakob** [comment]
Excellent, thank you!

**Pull request merged**